### PR TITLE
Add note `delete` feature

### PR DESCRIPTION
**Pull request by aliciamichellew**
Changes:
- Add note `delete` command feature 
- Fix small typo in Add note class
- Add exception description

### CODE REVIEW

File: src/main/java/seedu/address/model/notes/exceptions/NoteNotFoundException.java [hunk 1]
```diff
@@ -4,4 +4,7 @@
  * Signals that the operation is unable to find the specified note.
  */
 public class NoteNotFoundException extends RuntimeException {
+    public NoteNotFoundException() {
+        super("Note is not found");
```
Comment (dlathyun) [inline src/main/java/seedu/address/model/notes/exceptions/NoteNotFoundException.java:8]: Maybe the error message could be more specific, such as the note index is more than the size of note list?

File: src/main/java/seedu/address/model/notes/UniqueNoteList.java [hunk 2]
```diff
@@ -73,11 +73,21 @@ public void setNote(Note target, Note editedNote) {
...
+
+        for (int i = toRemove; i < internalList.size(); i++) {
+            internalList.get(i).setIndex(i);
+        }
+
+        Note deletedNote;
+        try {
+            deletedNote = internalList.remove(toRemove - 1);
+        } catch (Exception e) {
             throw new NoteNotFoundException();
```
Comment (dlathyun) [inline src/main/java/seedu/address/model/notes/UniqueNoteList.java:87]: Nice!

File: src/main/java/seedu/address/model/SeplendidModelManager.java [hunk 3]
```diff
@@ -449,8 +449,8 @@ public boolean hasNote(Note note) {
     }
 
     @Override
-    public void deleteNote(Note target) {
-        noteCatalogue.removeNote(target);
+    public Note deleteNote(int target) {
+        return noteCatalogue.removeNote(target);
```
Comment (dlathyun) [inline src/main/java/seedu/address/model/SeplendidModelManager.java:453]: Would it be necessary to call the updateNoteList to update the note list to show to the user?

**Review by dlathyun — CHANGES_REQUESTED** [review]
LGTM except I am not sure whether the displayed list will be updated after deleting the note!

File: src/main/java/seedu/address/model/SeplendidModelManager.java (hunk 3 above)
Comment (aliciamichellew) [reply src/main/java/seedu/address/model/SeplendidModelManager.java:453]: I don't think it's required. During my testing, the list is updated accordingly when I delete note.

Comment (dlathyun) [reply src/main/java/seedu/address/model/SeplendidModelManager.java:453]: Okay! I also just realised that after trying out.. Not sure why it doesn't work on some methods though. 👎

File: src/main/java/seedu/address/model/notes/exceptions/NoteNotFoundException.java [hunk 4]
```diff
@@ -5,6 +5,6 @@
  */
 public class NoteNotFoundException extends RuntimeException {
     public NoteNotFoundException() {
-        super("Note is not found");
+        super("Note is not found. Please check if the index you input sis valid.");
```
Comment (dlathyun) [inline src/main/java/seedu/address/model/notes/exceptions/NoteNotFoundException.java:8]: is*

**Review by dlathyun — APPROVED** [review]

**Pull request merged**